src/main/native/windows/process.cc

```diff
@@ -50,14 +50,18 @@ static bool DupeHandle(HANDLE h, AutoHandle* out, std::wstring* error) {
 bool WaitableProcess::Create(const std::wstring& argv0,
                              const std::wstring& argv_rest, void* env,
                              const std::wstring& wcwd, std::wstring* error) {
-  AutoHandle dup_in, dup_out, dup_err;
-  if (!DupeHandle(GetStdHandle(STD_INPUT_HANDLE), &dup_in, error) ||
-      !DupeHandle(GetStdHandle(STD_OUTPUT_HANDLE), &dup_out, error) ||
-      !DupeHandle(GetStdHandle(STD_ERROR_HANDLE), &dup_err, error)) {
-    return false;
-  }
-  return Create(argv0, argv_rest, env, wcwd, dup_in, dup_out, dup_err, nullptr,
-                true, error);
+  // On Windows 7, the subprocess cannot inherit console handles via the
+  // attribute list (CreateProcessW fails with ERROR_NO_SYSTEM_RESOURCES).
+  // If we pass only invalid handles here, the Create method creates an
+  // AutoAttributeList with 0 handles and initializes the STARTUPINFOEXW as
+  // empty and disallowing handle inheritance. At the same time, CreateProcessW
+  // specifies neither CREATE_NEW_CONSOLE nor DETACHED_PROCESS, so the
+  // subprocess *does* inherit the console, which is exactly what we want, and
+  // it works on all supported Windows versions.
+  // Fixes https://github.com/bazelbuild/bazel/issues/8676
+  return Create(argv0, argv_rest, env, wcwd, INVALID_HANDLE_VALUE,
```

Comment:
Although looks like this solves the problem, but we also need to make sure this doesn't leak any file handles other than `stdin`, `stdout` and `stderr` into the subprocess.

Reply:
I verified that it doesn't:
- created a program that uses the exact same code to create a subprocess
- the program also opens a file before creating the process
- i observed the subprocess' handles in Process Explorer.

With the current code, the subprocess does NOT inherit the file handle, even though I marked the handle inheritable.

If I change the code and pass the file handle as the stdout for the subprocess, then the handle is inherited and shown by process explorer.

Reply:
Nice, thanks for the verifying!!